auth/src/main/java/org/jboss/pnc/bacon/auth/KeycloakClientImpl.java

```diff
@@ -248,4 +208,52 @@ private void sleepExponentially(int retries) {
             throw new RuntimeException(e);
         }
     }
+
+    private InputStream constructKeycloakSettings(
+            String realm,
+            String keycloakBaseUrl,
+            String client,
+            boolean user,
+            String secret,
+            boolean refresh) {
+
+        String settings = "{\n" + "  \"realm\": \"" + realm + "\"," + "\n" + "  \"auth-server-url\": \""
```

Comment:
Could this be a map of strings and then we Jsonize it? I have some PTSD with hand-made JSON string and dealing with special cases in RHPAM :)

Reply:
sure!